### DISCUSSION TITLE
Laravel Passport. Change personal access client when creating token

### DISCUSSION THREAD

**Discussion by Aibat1997** [laravel/framework#49977] [Ideas]
In my project I use multiple authentication, I create provider, guards and create 2 personal access clients and add to them my provider to the column provider. 
![guards](https://github.com/laravel/framework/assets/34008555/581adf13-6cfd-45c7-a711-092088e3db47)

![access](https://github.com/laravel/framework/assets/34008555/d60b4f28-89e3-487e-ae8c-dc3a039a4e72)

But when I use Auth::guard('api-business')->user() it gets me instanse of User model not BusinessUser, to solve this I add to login 

```
$personal_access_client =  Passport::client()->where('provider', 'business_users')->where('personal_access_client', 1)->first();
Passport::useClientModel($personal_access_client);
```

and in vendor\laravel\passport\src\PersonalAccessTokenFactory.php I changed row 72:

```
$response = $this->dispatchRequestToAuthorizationServer(
            $this->createRequest(Passport::clientModel(), $userId, $scopes)
);
```

**Comment by hafezdivandari** [comment]
Fixed on Passport 13.x via https://github.com/laravel/passport/pull/1768
